draft-reddy-ipsecme-ikev2-pqc-auth.md

```diff
@@ -131,6 +132,53 @@ SLH-DSA does not introduce a new hardness assumption beyond those inherent to th
 
 # Signature Algorithm Use and Hashing in IKEv2 with ML-DSA and SLH-DSA
 
+For integrating ML-DSA and SLH-DSA into IKEv2, we take the approach used in [RFC8420]
+
+The implementation MUST send a SIGNATURE_HASH_ALGORITHMS notify with an Identity" (5) hash function.
+ML-DSA and SLH-DSA are only defined with the "Identity" hash and MUST NOT be sent to a receiver that has not indicated support for the "Identity" hash.
+
+When generating a signature with ML-DSA or SLH-DSA, the IKEv2 implementation would take the InitiatorSignedOctets string or the ResponderSignedOctets string (as appropriate), logically send it to the identity hash (which leaves it unchanged), and then pass it into the ML-DSA or SLH-DSA signer as the message to be signed (along with the fixed context string "IKEv2 Auth" (`49 4b 45 76 32 20 41 55 54 48`).
+The resulting signature is placed into the The resulting signature is the Signature Value.
+
+When verifying a signature with ML-DSA or SLH-DSA, the IKEv2 implementation would take the InitiatorSignedOctets string or the ResponderSignedOctets string (as appropriate), logically send it to the identity hash (which leaves it unchanged), and then pass it into the ML-DSA or SLH-DSA signer as the message to be verified (along with the fixed context string "IKEv2 Auth".
+
+## Discuession of ML-DSA and SLH-DSA and Prehashing
+
+The signature architecture within IKE was designed around RSA (and later extended to ECDSA).
+In this architecture, the actual message (the SignedOctets) are first hashed (using a hash that the verifier has indicated support for), and then passed for the remaining part of the signature generation processing.
+That is, it is designed for signature algorithms that first apply one of a number of hash functions to the message and then perform processing on that hash.
+Neither ML-DSA nor SLH-DSA fits cleanly into this architecture.
+
+We see three ways to address this mismatch.
+
+The first is to note that both ML-DSA and SLH-DSA have prehashed parameter sets; that is, ones designed to sign a message that has been hashed by an external source.
+At first place, this would appear to be an ideal solution, however it turns out that there are a number of practical issues.
+The first is that the prehashed version of ML-DSA and SLH-DSA would appear to be rarely used, and so it is not unlikely that support for it within crypto libraries may be lacking.
```

Comment:
More importantly, ML-DSA in X.509 does not support the pre-hash mode (see https://www.ietf.org/archive/id/draft-ietf-lamps-dilithium-certificates-07.html#section-8.1).

Reply:
\Actually, the lamps draft above is ambiguous (it can't be used for "PKIX applications, such as TLS" but they might be used for "end entity certificates").  I don't know where an IKE certificate would fall between the two - since we're not going that way anyways, there's no reason to open that can of worms.